main/autogen/cache.cc

```diff
@@ -13,6 +13,13 @@ namespace sorbet::autogen {
 
 const size_t MAX_SKIP_AMOUNT = 100;
 
+string_view trimPath(string_view path) {
+    if (path.size() > 2 && path[0] == '.' && path[1] == '/') {
+        return path.substr(2);
+    }
+    return path;
+}
+
```

Comment:
Usually checking for this prefix is a smell. Do you want to call `GlobalState::getPrintablePath`?